handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -0,0 +1,1951 @@
+                    }
+                }
+            }
+            return enabled;
+        }
+    }
+
+    @Override
+    public final void setEnabledCipherSuites(String[] cipherSuites) {
+        checkNotNull(cipherSuites, "cipherSuites");
+
+        final StringBuilder buf = new StringBuilder();
+        for (String c: cipherSuites) {
+            if (c == null) {
+                break;
+            }
+
+            String converted = CipherSuiteConverter.toOpenSsl(c);
+            if (converted == null) {
+                converted = c;
+            }
+
+            if (!OpenSsl.isCipherSuiteAvailable(converted)) {
+                throw new IllegalArgumentException("unsupported cipher suite: " + c + '(' + converted + ')');
+            }
+
+            buf.append(converted);
+            buf.append(':');
+        }
+
+        if (buf.length() == 0) {
+            throw new IllegalArgumentException("empty cipher suites");
+        }
+        buf.setLength(buf.length() - 1);
+
+        final String cipherSuiteSpec = buf.toString();
+
+        synchronized (this) {
+            if (!isDestroyed()) {
+                try {
+                    SSL.setCipherSuites(ssl, cipherSuiteSpec);
+                } catch (Exception e) {
+                    throw new IllegalStateException("failed to enable cipher suites: " + cipherSuiteSpec, e);
+                }
+            } else {
+                throw new IllegalStateException("failed to enable cipher suites: " + cipherSuiteSpec);
+            }
+        }
+    }
+
+    @Override
+    public final String[] getSupportedProtocols() {
+        return OpenSsl.SUPPORTED_PROTOCOLS_SET.toArray(new String[OpenSsl.SUPPORTED_PROTOCOLS_SET.size()]);
+    }
+
+    @Override
+    public final String[] getEnabledProtocols() {
+        List<String> enabled = InternalThreadLocalMap.get().arrayList();
+        // Seems like there is no way to explict disable SSLv2Hello in openssl so it is always enabled
+        enabled.add(OpenSsl.PROTOCOL_SSL_V2_HELLO);
+
+        int opts;
+        synchronized (this) {
+            if (!isDestroyed()) {
+                opts = SSL.getOptions(ssl);
+            } else {
+                return enabled.toArray(new String[1]);
+            }
+        }
+        if ((opts & SSL.SSL_OP_NO_TLSv1) == 0) {
+            enabled.add(OpenSsl.PROTOCOL_TLS_V1);
+        }
+        if ((opts & SSL.SSL_OP_NO_TLSv1_1) == 0) {
+            enabled.add(OpenSsl.PROTOCOL_TLS_V1_1);
+        }
+        if ((opts & SSL.SSL_OP_NO_TLSv1_2) == 0) {
+            enabled.add(OpenSsl.PROTOCOL_TLS_V1_2);
+        }
+        if ((opts & SSL.SSL_OP_NO_SSLv2) == 0) {
+            enabled.add(OpenSsl.PROTOCOL_SSL_V2);
+        }
+        if ((opts & SSL.SSL_OP_NO_SSLv3) == 0) {
+            enabled.add(OpenSsl.PROTOCOL_SSL_V3);
+        }
+        return enabled.toArray(new String[enabled.size()]);
+    }
+
+    @Override
+    public final void setEnabledProtocols(String[] protocols) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)